app/controllers/hyrax/downloads_controller.rb

```diff
@@ -29,7 +31,9 @@ def show
     # OVERRIDE Hyrax 2.9.6 for IRUS Analytics
     def item_identifier_for_irus_analytics
       # return the OAI identifier
-      "#{CatalogController.blacklight_config.oai[:provider][:record_prefix].call(self)}:#{params[:id]}"
+      # We must send the parent work id to IRUS otherwise it gets confused
+      parent_work_id = helpers.work_id_from_file_set_id(params[:id])
```

Comment:
What if you passed the controller and the id to the helper?  Then you could encapsulate the duplicate configuration logic.